### PR TITLE
GODRIVER-2728: Implement automatic Azure token acquisition callback

### PR DESCRIPTION
**Pull request by pmeredit**
## Summary

Implement automatic Azure token acquisition callback
## Background & Motivation

required by OIDC epic

### CODE REVIEW

**Comment by pmeredit** [comment]
@qingyang-hu sorry about the bot request, I was waiting for static analysis before adding reviewers

File: x/mongo/driver/auth/oidc.go [hunk 1]
```diff
@@ -178,6 +186,53 @@ func (oa *OIDCAuthenticator) providerCallback() (OIDCCallback, error) {
...
+		if resp.StatusCode != http.StatusOK {
+			return nil, newAuthError(fmt.Sprintf("failed to get a valid response from Azure Identity Provider, http code: %d", resp.StatusCode), nil)
+		}
+		err = json.NewDecoder(resp.Body).Decode(&azureResp)
+		if err != nil {
+			return nil, newAuthError("failed parsing result from Azure Identity Provider", err)
+		}
+		expiresOn, err := strconv.ParseInt(azureResp.ExpiresOn, 10, 64)
+		if err != nil {
+			return nil, newAuthError("failed converting experiation field from Azure Identity Provider to int64", err)
```
Comment (pmeredit) [inline x/mongo/driver/auth/oidc.go:226]:
```suggestion
			return nil, newAuthError("failed converting experation field from Azure Identity Provider to int64", err)
```

File: x/mongo/driver/auth/oidc.go [hunk 2]
```diff
@@ -178,6 +186,53 @@ func (oa *OIDCAuthenticator) providerCallback() (OIDCCallback, error) {
...
+		req.Header.Add("Accept", "application/json")
+		resp, err := httpClient.Do(req)
+		if err != nil {
+			return nil, newAuthError("error getting access token from Azure Identity Provider", err)
+		}
+		defer resp.Body.Close()
+		var azureResp struct {
+			AccessToken string `json:"access_token"`
+			ExpiresOn   string `json:"expires_on"`
+		}
```
Comment (matthewdale) [inline x/mongo/driver/auth/oidc.go:215]: Optional: You can skip the separate integer parsing step by using the `,string` struct tag option for `ExpiresOn`.

E.g.
```go
var azureResp struct {
	AccessToken string `json:"access_token"`
	ExpiresOn   int64  `json:"expires_on,string"`
}
```

Comment (pmeredit) [reply x/mongo/driver/auth/oidc.go:215]: Oh, cool!

**Review by matthewdale — APPROVED** [review]
Looks good! 👍

**Pull request merged**